src/tests/end-to-end/tests/details-view/preview-features.test.ts

```diff
@@ -76,7 +76,7 @@ describe('Preview Features Panel', () => {
 
         const detailsViewPage = await waitForDetailsViewWithPreviewFeaturesPanel(browser, popupPage, targetTabId);
 
-        await detailsViewPage.waitForSelector(detailsViewSelectors.noPreviewFeaturesMessage);
+        await detailsViewPage.waitForSelector(detailsViewSelectors.previewFeaturesPanel);
```

Comment:
This is an important change that I'm open to discussing.

Since there are some E2E tests that are dependent on css classnames, it might be a good idea to think about the pattern in which we do this. Should we start adding IDs for end to end tests, since the generated css selectors will not be something that jest can reproduce* (since it is not dependent on compiled code); the IDs would not be used for CSS but only for E2E tests, which might not be ideal for some.

Alternatively, we can also not use wait for selector and rely on another API.

And finally, there is also using this: https://www.npmjs.com/package/jest-transform-css.

Personally, I think the first solution is the easiest and would allow us to move forward without investing too much time.

* this is due to the suggested guidance on for using jest with CSS modules: https://jestjs.io/docs/en/webpack#mocking-css-modules.

Reply:
I think there is also something called `automation-id` which is similar to adding `id` but that is specific to our automated e2e tests. We can also check those and how that can fit in our e2e. Just my 2 cents here.

Reply:
I like option 1 (just using id). automation-id is fine if there are id conflicts but I don't really see a reason to prefer it over id for the cases where id would work

Reply:
There's also the possibility of continuing to specify the CSS class in the test code as a string (i.e. not using CSS modules to get the class name), yes? Or does CSS modules munge the class name?

Reply:
@peterdur unfortunately there is no way to do this for E2E tests. This is because E2E tests depend on the produced bundle code when comparing screenshots VERSUS Jest which will be relying on `identity-object-proxy` to get the class-names, so in that scenario there would be soft-coupling.

Admittedly, we already have this but ideally we could improve this by relying on IDs that are accessible and guaranteed to be the same.